homeassistant/components/roborock/number.py

```diff
@@ -57,22 +60,37 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up Roborock number platform."""
-
     coordinators: dict[str, RoborockDataUpdateCoordinator] = hass.data[DOMAIN][
         config_entry.entry_id
     ]
-    async_add_entities(
-        (
-            RoborockNumberEntity(
-                f"{description.key}_{slugify(device_id)}",
-                coordinator,
-                description,
-            )
-            for device_id, coordinator in coordinators.items()
-            for description in NUMBER_DESCRIPTIONS
+    possible_entities: list[
+        tuple[RoborockDataUpdateCoordinator, RoborockNumberDescription]
+    ] = [
+        (coordinator, description)
+        for coordinator in coordinators.values()
+        for description in NUMBER_DESCRIPTIONS
+    ]
+    # We need to check if this function is supported by the device.
+    results = await asyncio.gather(
+        *(
+            coordinator.api.cache.get(description.cache_key).async_value()
+            for coordinator, description in possible_entities
         ),
-        True,
+        return_exceptions=True,
     )
+    valid_entities: list[RoborockNumberEntity] = []
+    for (coordinator, description), result in zip(possible_entities, results):
+        if result is None or isinstance(result, Exception):
```

Comment:
The standard is not to catch plain exceptions, but to have more specific exceptions to avoid accidental bugs.

When does this happen? Is it expected to silently omit entities?

Reply:
Well I have return exceptions=true.
Sk ok any random exception that occurs will be passed through instead of crashing there - we notify the user that something went wrong when adding that entity, I.e. it isn't supported, or something about their device made the code crash.

I could potentially add more if statements for different errors it could be, but since I'm doing return exceptions, I still have to handle all exceptions

Reply:
Just to clarify, i'm specifically talking about catching plain `Exception` vs more specific exceptions. That is, my comment is not about `return_exceptions` and applies even if this were a normal try / except block.  I am imaginign the code is the same as:

```
try:
    ...
except Exception:
```
which we wouldn't allow here.

Reply:
But it isn't quite as simple as that, as right now return_exceptions basically does do 

```
try:
    return function call
except Exception:
    return exception
```

I'm just changing that to log rather than to do nothing. I can change it to RoborockException instead of Exception though if you would like, but I still have to make sure that it is returning a valid response and not an exception

Reply:
I think it's worth the change to RoborockException. It's not super specific, but at least it tells readers we are expecting an error from roborock and not elsewhere, if that makes sense

Reply:
Sounds good - changed!